File: Encontro.Domain/Entities/EventParticipant.cs
using System.ComponentModel.DataAnnotations;
using Encontro.Domain.Interfaces;

namespace Encontro.Domain.Entities;

public class EventParticipant : ISoftDeletable
{
    public int Id { get; set; }

    [Required(ErrorMessage = "O evento é obrigatório")]
    [Display(Name = "Evento")]
    public int EventId { get; set; }
    public Event Event { get; set; } = null!;

    [Required(ErrorMessage = "A pessoa é obrigatória")]
    [Display(Name = "Pessoa")]
    public int PersonId { get; set; }
    public Person Person { get; set; } = null!;

    [Display(Name = "Equipe")]
    public int? TeamId { get; set; }
    public Team? Team { get; set; }

    [Display(Name = "Função")]
    public int? RoleId { get; set; }
    public Role? Role { get; set; }

    [Display(Name = "Data de Inscrição")]
    public DateTime RegisteredAt { get; set; }

    [Required(ErrorMessage = "A etapa é obrigatória")]
    [Range(1, 10, ErrorMessage = "A etapa deve estar entre 1 e 10")]
    [Display(Name = "Etapa")]
    public int Stage { get; set; }

    [MaxLength(500, ErrorMessage = "As observações devem ter no máximo 500 caracteres")]
    [Display(Name = "Observações")]
    public string? Notes { get; set; }

    [Display(Name = "Indicado")]
    public bool? Indicated { get; set; }

    [Display(Name = "Aceitou")]
    public bool? Accepted { get; set; }

    // Soft Delete
    public bool IsDeleted { get; set; }
    public DateTime? DeletedAt { get; set; }
    public string? DeletedBy { get; set; }
}

File: app/Models/EventParticipant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class EventParticipant extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'event_id',
        'person_id',
        'team_id',
        'role_id',
        'registered_at',
        'stage',
        'notes',
        'indicated',
        'accepted',
        'is_deleted',
        'deleted_by',
    ];

    protected $casts = [
        'registered_at' => 'datetime',
        'indicated' => 'boolean',
        'accepted' => 'boolean',
        'is_deleted' => 'boolean',
    ];

    public function event()
    {
        return $this->belongsTo(Event::class);
    }

    public function person()
    {
        return $this->belongsTo(Person::class);
    }

    public function team()
    {
        return $this->belongsTo(Team::class);
    }

    public function role()
    {
        return $this->belongsTo(Role::class);
    }
}
